src/graph/DeleteEdgesExecutor.cpp

```diff
@@ -20,42 +19,44 @@ DeleteEdgesExecutor::DeleteEdgesExecutor(Sentence *sentence,
 }
 
 Status DeleteEdgesExecutor::prepare() {
-    spaceId_ = ectx()->rctx()->session()->space();
-    expCtx_ = std::make_unique<ExpressionContext>();
-    expCtx_->setSpace(spaceId_);
-    expCtx_->setStorageClient(ectx()->getStorageClient());
-
-    auto edgeStatus = ectx()->schemaManager()->toEdgeType(spaceId_, *sentence_->edge());
-    if (!edgeStatus.ok()) {
-        return edgeStatus.status();
-    } else {
-        edgeType_ = edgeStatus.value();
-        auto schema = ectx()->schemaManager()->getEdgeSchema(spaceId_, edgeType_);
-        if (schema == nullptr) {
-            return Status::Error("No schema found for '%s'", sentence_->edge()->c_str());
-        }
-    }
-
     return Status::OK();
 }
 
 void DeleteEdgesExecutor::execute() {
     auto status = checkIfGraphSpaceChosen();
     if (!status.ok()) {
-        DCHECK(onError_);
-        onError_(std::move(status));
+        doError(std::move(status));
         return;
     }
 
-    status = setupEdgeKeys();
+    auto space = ectx()->rctx()->session()->space();
```

Comment:
What's the purpose about the changes in DeleteEdgesExecutor.cpp .  I can't see any difference

Reply:
We shouldn't check it in prepare step, for example; `Use space; delete ...`

Reply:
why?  Is there any difference?  You could move checkIfGraphSpaceChosen into prepare.

Reply:
why?  Is there any difference?  You could move checkIfGraphSpaceChosen into prepare step.

Reply:
For example:

```
(user@127.0.0.1:13699) [t]> INSERT VERTEX player(name, age) VALUES 100:("Tim Duncan", 42)
Execution succeeded (Time spent: 2.233/3.439 ms)

Fri Feb 14 19:45:25 2020

(user@127.0.0.1:13699) [t]> FETCH PROP ON player 100;
=======================================
| VertexID | player.name | player.age |
=======================================
| 100      | Tim Duncan  | 42         |
---------------------------------------
Got 1 rows (Time spent: 5.353/6.825 ms)

Fri Feb 14 19:45:27 2020

(user@127.0.0.1:13699) [t]> use test
Execution succeeded (Time spent: 1.81/3.018 ms)

Fri Feb 14 19:45:32 2020

(user@127.0.0.1:13699) [test]> use t ; DELETE VERTEX 100;
Execution succeeded (Time spent: 3.388/4.067 ms)

Fri Feb 14 19:45:37 2020

(user@127.0.0.1:13699) [t]> FETCH PROP ON player 100;
=======================================
| VertexID | player.name | player.age |
=======================================
| 100      | Tim Duncan  | 42         |
---------------------------------------
Got 1 rows (Time spent: 3.496/4.686 ms)

Fri Feb 14 19:45:40 2020
```